gremlin-go/driver/result.go

```diff
@@ -147,58 +147,74 @@ func (r *Result) IsNil() bool {
 
 // GetVertex returns the result if it is a vertex, otherwise returns an error
 func (r *Result) GetVertex() (*Vertex, error) {
-	res, ok := r.result.(Vertex)
+	res, ok := r.result.(*Vertex)
 	if !ok {
 		return nil, errors.New("result is not a Vertex")
 	}
-	return &res, nil
+	return res, nil
 }
 
 // GetEdge returns the result if it is an edge, otherwise returns an error
 func (r *Result) GetEdge() (*Edge, error) {
-	res, ok := r.result.(Edge)
+	res, ok := r.result.(*Edge)
 	if !ok {
 		return nil, errors.New("result is not an Edge")
 	}
-	return &res, nil
+	return res, nil
 }
 
 // GetElement returns the result if it is an element, otherwise returns an error
 func (r *Result) GetElement() (*Element, error) {
-	res, ok := r.result.(Element)
+	res, ok := r.result.(*Element)
 	if !ok {
 		return nil, errors.New("result is not an Element")
 	}
-	return &res, nil
+	return res, nil
 }
 
 // GetPath returns the result if it is a path, otherwise returns an error
 func (r *Result) GetPath() (*Path, error) {
-	res, ok := r.result.(Path)
+	res, ok := r.result.(*Path)
 	if !ok {
 		return nil, errors.New("result is not a Path")
 	}
-	return &res, nil
+	return res, nil
 }
 
 // GetProperty returns the result if it is a property, otherwise returns an error
 func (r *Result) GetProperty() (*Property, error) {
-	res, ok := r.result.(Property)
+	res, ok := r.result.(*Property)
 	if !ok {
 		return nil, errors.New("result is not a Property")
 	}
-	return &res, nil
+	return res, nil
 }
 
 // GetVertexProperty returns the result if it is a vertex property, otherwise returns an error
 func (r *Result) GetVertexProperty() (*VertexProperty, error) {
-	res, ok := r.result.(VertexProperty)
+	res, ok := r.result.(*VertexProperty)
 	if !ok {
 		return nil, errors.New("result is not a VertexProperty")
 	}
+	return res, nil
+}
+
+func (r *Result) GetTraverser() (*Traverser, error) {
+	res, ok := r.result.(Traverser)
+	if !ok {
+		return nil, errors.New("result is not a Traverser")
+	}
 	return &res, nil
```

Comment:
Nit:
For this and GetSLice the return value of res seems to be inconsistent with the rest of the project, where it is cast to a pointer instead of the value.

Is this intentional or an oversight, and if it's intentional can it be adjusted for consistency?

Reply:
Fixed. GetSlice is intentional and hard to change. Traverser has been changed to be same style.